app/screens/login/login.js

```diff
@@ -73,18 +74,25 @@ export default class Login extends PureComponent {
         setMfaPreflightDone(false);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.loginRequest.status === RequestStatus.STARTED && nextProps.loginRequest.status === RequestStatus.SUCCESS) {
-            this.props.actions.handleSuccessfulLogin().then(this.goToChannel);
-        } else if (this.props.loginRequest.status !== nextProps.loginRequest.status && nextProps.loginRequest.status !== RequestStatus.STARTED) {
-            this.setState({isLoading: false});
+    componentDidUpdate(nextProps) {
+        const {loginRequest: {status}, actions: {handleSuccessfulLogin}} = this.props;
+
+        if (status === RequestStatus.STARTED && nextProps.loginRequest.status === RequestStatus.SUCCESS) {
```

Comment:
This is making it so that you're sent to the channel when the request status goes from successful to in-progress which isn't correct. This is probably why the unit test is failing.